packages/react-data-grid/src/ReactDataGrid.tsx

```diff
@@ -0,0 +1,729 @@
+  onRowDoubleClick?(rowIdx: number, rowData: RowData, column: Column): void;
+  onAddFilter?(args: FilterArgs): void;
+  onClearFilters?(): void;
+  /** Function called whenever grid is sorted*/
+  onGridSort?: GridProps['onSort'];
+  /** Function called whenever keyboard key is pressed up */
+  onGridKeyUp?(e: React.KeyboardEvent<HTMLDivElement>): void;
+  /** Function called whenever keyboard key is pressed down */
+  onGridKeyDown?(e: React.KeyboardEvent<HTMLDivElement>): void;
+  onRowSelect?(rowData: RowData[]): void;
+  columnEquality(c1: Column, c2: Column): boolean;
+  rowSelection?: {
+    enableShiftSelect?: boolean;
+    /** Function called whenever rows are selected */
+    onRowsSelected?(args: ({ rowIdx: number; row: RowData })[]): void;
+    /** Function called whenever rows are deselected */
+    onRowsDeselected?(args: ({ rowIdx: number; row: RowData })[]): void;
+    /** toggle whether to show a checkbox in first column to select rows */
+    showCheckbox?: boolean;
+    /** Method by which rows should be selected */
+    selectBy: RowSelection;
+  };
+  /** Custom checkbox formatter */
+  rowActionsCell?: React.ComponentType<CheckboxEditorProps>;
+  /**
+   * Callback called whenever row data is updated
+   * When editing is enabled, this callback will be called for the following scenarios
+   * 1. Using the supplied editor of the column. The default editor is the [SimpleTextEditor](https://github.com/adazzle/react-data-grid/blob/master/packages/common/editors/SimpleTextEditor.js).
+   * 2. Copy/pasting the value from one cell to another <kbd>CTRL</kbd>+<kbd>C</kbd>, <kbd>CTRL</kbd>+<kbd>V</kbd>
+   * 3. Update multiple cells by dragging the fill handle of a cell up or down to a destination cell.
+   * 4. Update all cells under a given cell by double clicking the cell's fill handle.
+   */
+  onGridRowsUpdated?(args: {
+    cellKey: string;
+    fromRow: number;
+    toRow: number;
+    fromRowId: unknown;
+    toRowId: unknown;
+    rowIds: unknown[];
+    updated: { [key: string]: unknown };
+    action: UpdateActions;
+    fromRowData: RowData;
+  }): void;
+}
+
+interface State {
+  columnMetrics: ColumnMetrics;
+  lastRowIdxUiSelected: number;
+  selectedRows: RowData[];
+  canFilter?: boolean;
+  sortColumn?: string;
+  sortDirection?: DEFINE_SORT;
+}
+
+function isRowSelected(keys: unknown, indexes: unknown, isSelectedKey: unknown, rowData: RowData, rowIdx: number) {
+  return rowUtils.isRowSelected(keys as { rowKey?: string; values?: string[] } | null, indexes as number[] | null, isSelectedKey as string | null, rowData, rowIdx);
+}
+
+/** Main API Component to render a data grid of rows and columns
+ *
+ * Example code
+ * -----
+ *
+ * ```javascript
+ * <ReactDataGrid
+ *   columns={columns}
+ *   rowGetter={i => rows[i]}
+ *   rowsCount={3} />
+ * ```
+*/
+export default class ReactDataGrid extends React.Component<Props, State> {
+  static displayName = 'ReactDataGrid';
+
+  static defaultProps = {
```

Comment:
> The default-ed properties are inferred from the defaultProps property type. If an explicit type annotation is added, e.g. static defaultProps: Partial<Props>; the compiler will not be able to identify which properties have defaults (since the type of defaultProps include all properties of Props).

> Use static defaultProps: Pick<Props, "name">; as an explicit type annotation instead, or do not add a type annotation as done in the example above.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html

Do you want me to add `Pick<Props, "name">`?

Reply:
I added `DefaultProps` type